src/main/java/com/uw/dao/TraineeDaoImpl.java

```diff
@@ -1,43 +1,179 @@
     }
 
     @Override
     public void update(Trainee trainee) {
-        this.storage.save( trainee.getUserId(), trainee );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try{
+            transaction = session.beginTransaction();
+            session.merge(trainee);
+            transaction.commit();
+        }catch(Exception e){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public void delete(long id) {
-        this.storage.remove( id );
+        Session session = sessionFactory.openSession();
+        Transaction transaction = null;
+        try{
+            transaction = session.beginTransaction();
+            Trainee traineeToDelete = session.get(Trainee.class, id);
+            session.remove(traineeToDelete);
+            transaction.commit();
+        }catch(Exception e){
+            transaction.rollback();
+        }finally{
+            session.close();
+        }
     }
 
     @Override
     public Trainee selectProfile(long id) {
-        return ( Trainee ) this.storage.get( id );
+        Session session = this.sessionFactory.openSession();
+        Transaction transaction = null;
+        Trainee trainee = null;
+        try{
+            trainee = session.get (Trainee.class, id);
+        }catch ( Exception e){
+            e.printStackTrace();
+        }finally{
+            session.close();
+        }
+        return trainee;
+    }
+
+    @Override
+    public List<Trainee> findAll() {
+        Session session = sessionFactory.openSession();
+        List<Trainee> traineeList = null;
+
+        try {
+            // Crear un objeto Criteria para la clase de entidad User
+            Query query = session.createQuery("select t from Trainee t", Trainee.class);
+            traineeList = query.getResultList();
+        } catch (HibernateException e) {
+            e.printStackTrace();
+        } catch( NoResultException e){
+            e.printStackTrace();
+        }finally {
+            session.close();
+        }
+        return traineeList;
     }
 
-    public Storage getStorage() {
-        return this.storage;
+    @Override
+    public Trainee existTraineeByUserId(long id) {
+        Session session = sessionFactory.openSession();
+        Trainee trainee = null;
+        try{
+            org.hibernate.query.Query<Trainee> query = session.createQuery("select t from Trainee t where t.user.id = :id", Trainee.class).setParameter("id", id);
+            trainee = query.getSingleResult();
+        }catch ( NoResultException e ){
+            // Ignore this
+        }catch(Exception e){
+            e.printStackTrace();
+        }finally {
+            session.close();
+        }
+        return trainee;
     }
+
+    @Override
+    public Trainee findTraineeByUsername(String username) {
+        Trainee trainee = null;
+        try( Session session = sessionFactory.openSession()){
```

Comment:
Ohh you used here indeed, please make sure you standardize the code following this approach